README.md


<div align="center">

  ![Logo](https://user-images.githubusercontent.com/62848565/128122647-f56c9c5a-54ef-4f6e-b6ad-65964d345251.png)
  
</div>
<div align="center">


[![flutter](https://img.shields.io/badge/Flutter%20ver.-2.4.0-blue?style=for-the-badge&logo=flutter)](https://flutter.dev/)
[![GitHub license](https://img.shields.io/badge/license-MIT-red.svg?style=for-the-badge)](https://github.com/data-charya/Synth-Flutter/blob/master/LICENSE)
[![Flutter action](https://github.com/data-charya/Synth-Flutter/actions/workflows/dart.yml/badge.svg)](https://github.com/data-charya/Synth-Flutter/actions/workflows/dart.yml)

</div>
<hr>

# Synth Neumorphism Components

My take on CRED's Synth Design System. I tried to recreate the components as close to the real thing as possible and will be adding more components down the line.

<br>

## Usage
Add the plugin to your pubspec.yaml using :
```dart
flutter pub add synth
```
Import the package and use the components, Its that simple !
```dart
SynthSoftbutton(
  text: 'Your Text',
  width: 100,
  height: 50,
  onPressed: () 
  {
    //write your function definition
  },
 )
```
## Buttons
### Softbutton


![1](https://user-images.githubusercontent.com/62848565/128127176-0f50a578-1ee7-47a4-b67d-4f72a2f14161.png)

```dart
SynthSoftbutton(
  text: 'Your Text',
  width: 100,
  height: 50,
  onPressed: () 
  {
    //write your function definition
  },
 )
```

<br>

### Drawable Button


![2](https://user-images.githubusercontent.com/62848565/128127187-51f5ced1-3ee0-4976-80d6-fd885e0b72bf.png)

```dart
SynthDrawablebutton(
  text: 'Your Text',
  width: 100,
  height: 50,
  onPressed: () 
  {
    //write your function definition
  },
 )
```
<br>

### Flat Button
![3](https://user-images.githubusercontent.com/62848565/128127202-784ff6fb-dcd4-4099-a81d-0d6f64542158.png)

```dart
SynthFlatbutton(
  text: 'Your Text',
  width: 100,
  height: 50,
  onPressed: () 
  {
    //write your function definition
  },
 )
```

<br>

### Flat Drawable Button
![4](https://user-images.githubusercontent.com/62848565/128127223-5227e45a-62f7-4016-8e99-69aa1363a08b.png)

```dart
SynthFlatDrawablebutton(
  text: 'Your Text',
  width: 100,
  height: 50,
  onPressed: () 
  {
    //write your function definition
  },
  icon: Icon(
    Icons.add,
    color: primarySynthWhite100,
    ),
   )
 )
```

<br>

### Img Button
![5](https://user-images.githubusercontent.com/62848565/128127242-ec0c5bb6-3ae1-4531-b951-88ea34def291.png)

```dart
NeuImgbutton(
  radius: 80,
  onPressed: () 
  {
    //write your function definition
  },
  icon: Icon(
    Icons.arrow_back_ios_new_rounded,
  ),
 )
```

<br>

### Elevated View
![6](https://user-images.githubusercontent.com/62848565/128127250-52dbe753-cffb-4de2-af7a-a33af6ae5e55.png)

```dart
ElevatedView(
   width: 300,
   height: 100,
   child: Center(
   child: Text(
   'synth',
   style: TextStyle(
   fontSize: 25,
   fontFamily: 'Gilroy',
   fontWeight: FontWeight.w900,
   color: primarySynthCopper400,
   ),
  ),
 ),
)
```

<br>

### Circular Checkbox
![7](https://user-images.githubusercontent.com/62848565/128127263-e559d648-8689-4079-b908-36e359f10701.png)

```dart
SynthCircularcheckbox(
  width: 30,
  height: 30,
  onPressed: () 
  {
    //write your function definition
  },
 )
```

<br>

### RoundedRect Checkbox
![8](https://user-images.githubusercontent.com/62848565/128127271-3784936b-edd3-4e3a-85f1-daa3e886e4e0.png)

```dart
 SynthRoundedRectcheckbox(
   width: 30,
   height: 30,
   onPressed: () 
   {
      //write your function definition here
   },
  )
```

<br>


CHANGELOG.md
## 1.0.4

- Major final release.


test/synth_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:synth/synth.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
  });
}


example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:synth_example/main.dart';

void main() {
  testWidgets('Verify Platform version', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(const Synth());

    // Verify that platform version is retrieved.
    expect(
      find.byWidgetPredicate(
        (Widget widget) =>
            widget is Text && widget.data!.startsWith('Running on:'),
      ),
      findsOneWidget,
    );
  });
}


lib/src/color_page.dart
import 'dart:ui';

import 'package:flutter/cupertino.dart';

//Primary
//Synth Black
Color primaryBlack = const Color(0xff000000);
Color primarySynthBlack = const Color(0xff212426);
Color primarySynthBlack200 = const Color(0xff1B1E20);
Color primarySynthBlack400 = const Color(0xff171818);

//Synth White
Color primarySynthWhite = const Color(0xff7C7D7F);
Color primarySynthWhite100 = const Color(0xffFFFFFF);
Color primarySynthWhite200 = const Color(0xffCFD0D0);
Color primarySynthWhite300 = const Color(0xff4B4E4F);
Color primarySynthWhite400 = const Color(0xff4B4E4F);
Color primarySynthWhite500 = const Color(0xff343739);
Color primarySynthWhite600 = const Color(0xff2A2D2F);

//Synth Copper
Color primarySynthCopper40 = const Color(0xff7A6B61);
Color primarySynthCopper400 = const Color(0xffF9D3B4);
Color primarySynthCopper600 = const Color(0xffD9896A);
Color primarySynthCopper800 = const Color(0xff945439);
LinearGradient primarySynthCopperHorizontalGradient = const LinearGradient(
    colors: [Color(0xff95553A), Color(0xffD78869)],
    begin: Alignment.centerRight,
    end: Alignment.centerLeft);
LinearGradient primarySynthCopperVerticalGradient = const LinearGradient(
    colors: [Color(0xff96563A), Color(0xffD78869)],
    begin: Alignment.bottomCenter,
    end: Alignment.topCenter);

//Synth Blue
Color primarySynthBlue400 = const Color(0xff426BB6);
Color primarySynthBlue700 = const Color(0xff26458C);
Color primarySynthBlue800 = const Color(0xff1B2D5B);
Color primarySynthBlue900 = const Color(0xff18263E);

LinearGradient primarySynthBlueGradient = const LinearGradient(
  colors: [
    Color(0xff426BB6),
    Color(0xff26458C),
  ],
  begin: Alignment.topCenter,
  end: Alignment.bottomCenter,
);

LinearGradient primarySynthBlueDarkGradient = const LinearGradient(
  colors: [
    Color(0xff2C497F),
    Color(0xff2A4A92),
  ],
  begin: Alignment.topCenter,
  end: Alignment.bottomCenter,
);

//Secondary
//Synth Purple
Color secondarySynthPurple200 = const Color(0xffBB86FC);
Color secondarySynthPurple300 = const Color(0xff7A5A94);
Color secondarySynthPurple400 = const Color(0xff4C4266);
Color secondarySynthPurple500 = const Color(0xff6200EE);
Color secondarySynthPurple700 = const Color(0xff3700B3);

//Synth Teal
Color secondarySynthTeal200 = const Color(0xff018786);
Color secondarySynthTeal700 = const Color(0xff03DAC5);


android/src/main/kotlin/io/shanwill/synth/SynthPlugin.kt
package io.shanwill.synth

import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** SynthPlugin */
class SynthPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "synth")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    if (call.method == "getPlatformVersion") {
      result.success("Android ${android.os.Build.VERSION.RELEASE}")
    } else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:synth/synth.dart';

void main() {
  runApp(const Synth());
}

class Synth extends StatelessWidget {
  const Synth({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Synth',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        scaffoldBackgroundColor: primarySynthBlack,
      ),
      home: const Homepage(),
    );
  }
}

class Homepage extends StatefulWidget {
  const Homepage({Key? key}) : super(key: key);

  @override
  _HomepageState createState() => _HomepageState();
}

class _HomepageState extends State<Homepage> {
  @override
  Widget build(BuildContext context) {
    var size = MediaQuery.of(context).size;
    return SafeArea(
      child: Scaffold(
        body: SizedBox(
          width: size.width,
          height: size.height,
          child: ListView(
            scrollDirection: Axis.vertical,
            children: [
              Row(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  Text(
                    'synth',
                    style: TextStyle(
                      fontSize: 65,
                      fontFamily: 'Gilroy',
                      fontWeight: FontWeight.w900,
                      color: primarySynthCopper400,
                    ),
                  ),
                ],
              ),
              Padding(
                padding: const EdgeInsets.symmetric(vertical: 40),
                child: Column(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: [
                    SynthSoftbutton(
                      text: 'Agree & continue',
                      width: 200,
                      height: 60,
                      onPressed: () {},
                    ),
                    Padding(
                      padding: const EdgeInsets.only(top: 40),
                      child: SynthDrawablebutton(
                          text: 'Idle',
                          width: 180,
                          height: 80,
                          onPressed: () {}),
                    )
                  ],
                ),
              ),
              Padding(
                padding: const EdgeInsets.symmetric(vertical: 40),
                child: Column(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: [
                    SynthFlatbutton(
                      text: 'Idle',
                      width: 180,
                      height: 80,
                      onPressed: () {},
                      icon: const Icon(
                        Icons.add,
                        color: Colors.white,
                      ),
                    ),
                    Padding(
                      padding: const EdgeInsets.only(top: 30),
                      child: SynthFlatDrawablebutton(
                        text: 'Idle',
                        width: 180,
                        height: 80,
                        onPressed: () {},
                        icon: Icon(
                          Icons.add,
                          color: primarySynthWhite100,
                        ),
                      ),
                    ),
                  ],
                ),
              ),
              Padding(
                padding: const EdgeInsets.symmetric(vertical: 40),
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: [
                    NeuImgbutton(
                      radius: 80,
                      onPressed: () {},
                      icon: const Icon(
                        Icons.arrow_back_ios_new_rounded,
                        color: Colors.white,
                      ),
                    ),
                    NeuImgbutton(
                      radius: 80,
                      onPressed: () {},
                      icon: const Icon(
                        Icons.arrow_forward_ios_rounded,
                        color: Colors.white,
                      ),
                    ),
                  ],
                ),
              ),
              Padding(
                padding: const EdgeInsets.symmetric(vertical: 30),
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    ElevatedView(
                      width: 300,
                      height: 100,
                      child: Center(
                        child: Text(
                          'synth',
                          style: TextStyle(
                            fontSize: 25,
                            fontFamily: 'Gilroy',
                            fontWeight: FontWeight.w900,
                            color: primarySynthCopper400,
                          ),
                        ),
                      ),
                    ),
                  ],
                ),
              ),
              Padding(
                padding: const EdgeInsets.symmetric(vertical: 20),
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    SynthCircularcheckbox(
                      width: 30,
                      height: 30,
                      onPressed: () {},
                    ),
                    Padding(
                      padding: const EdgeInsets.only(left: 30),
                      child: SynthRoundedRectcheckbox(
                        width: 30,
                        height: 30,
                        onPressed: () {},
                      ),
                    )
                  ],
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
